L1Trigger/CSCTriggerPrimitives/src/CSCUpgradeMotherboardLUT.cc

```diff
@@ -0,0 +1,256 @@
+#include "L1Trigger/CSCTriggerPrimitives/src/CSCUpgradeMotherboardLUT.h"
+
+CSCGEMMotherboardLUT::CSCGEMMotherboardLUT() 
+  : lut_wg_eta_odd(0)
+  , lut_wg_eta_even(0)
+  , lut_pt_vs_dphi_gemcsc(0)
+
+  , gem_roll_eta_limits_odd_l1(0)
+  , gem_roll_eta_limits_odd_l2(0)
+  , gem_roll_eta_limits_even_l1(0)
+  , gem_roll_eta_limits_even_l2(0)
+    
+  , csc_wg_to_gem_roll_odd_l1(0)
+  , csc_wg_to_gem_roll_odd_l2(0)
+  , csc_wg_to_gem_roll_even_l1(0) 
+  , csc_wg_to_gem_roll_even_l2(0)
+{
+}
+
+std::vector<std::pair<int,int> >
+CSCGEMMotherboardLUT::get_csc_wg_to_gem_roll(Parity par, int layer) const
+{
+  if (par==Parity::Even){ return layer==1 ? csc_wg_to_gem_roll_even_l1 : csc_wg_to_gem_roll_even_l2; }
+  else                  { return layer==1 ? csc_wg_to_gem_roll_odd_l1 :  csc_wg_to_gem_roll_odd_l2; }
+} 
+
+std::vector<int>
+CSCGEMMotherboardLUT::get_gem_roll_to_csc_wg(Parity par, int layer) const
+{
+  if (par==Parity::Even){ return layer==1 ? gem_roll_to_csc_wg_even_l1 : gem_roll_to_csc_wg_even_l2; }
+  else                  { return layer==1 ? gem_roll_to_csc_wg_odd_l1 :  gem_roll_to_csc_wg_odd_l2; }
+}
+
+std::vector<int> 
+CSCGEMMotherboardLUTME11::get_gem_pad_to_csc_hs(Parity par, enum CSCPart p) const
+{
+  if (p==CSCPart::ME1A) { return par==Parity::Even ? gem_pad_to_csc_hs_me1a_even : gem_pad_to_csc_hs_me1a_odd; }
+  else                  { return par==Parity::Even ? gem_pad_to_csc_hs_me1b_even : gem_pad_to_csc_hs_me1b_odd; }
+}
+
+std::vector<int> 
+CSCGEMMotherboardLUTME21::get_gem_pad_to_csc_hs(Parity par, enum CSCPart p) const
+{
+  return par==Parity::Even ? gem_pad_to_csc_hs_even : gem_pad_to_csc_hs_odd;
+}
+
+std::vector<std::pair<int,int> > 
+CSCGEMMotherboardLUTME21::get_csc_hs_to_gem_pad(Parity par, enum CSCPart p) const
+{
+  return par==Parity::Even ? csc_hs_to_gem_pad_even : csc_hs_to_gem_pad_odd;
+}
+
+std::vector<std::pair<int,int> > 
+CSCGEMMotherboardLUTME11::get_csc_hs_to_gem_pad(Parity par, enum CSCPart p) const
+{
+  if (p==CSCPart::ME1A) { return par==Parity::Even ? csc_hs_to_gem_pad_me1a_even : csc_hs_to_gem_pad_me1a_odd; }
+  else                  { return par==Parity::Even ? csc_hs_to_gem_pad_me1b_even : csc_hs_to_gem_pad_me1b_odd; }
+}
+
+std::vector<std::vector<double> > 
+CSCGEMMotherboardLUTME11::get_lut_wg_vs_hs(enum CSCPart p) const
+{
+  if (p==CSCPart::ME1A)      { return lut_wg_vs_hs_me1a;  }
+  else if (p==CSCPart::ME1B) { return lut_wg_vs_hs_me1b;  }
+  else                       { return lut_wg_vs_hs_me1ag; } 
+}
+
+CSCGEMMotherboardLUT::~CSCGEMMotherboardLUT()
+{
+}
+
+
+CSCGEMMotherboardLUTME11::CSCGEMMotherboardLUTME11() 
+  : CSCGEMMotherboardLUT()
+{
+  lut_wg_eta_odd = {};
+  lut_wg_eta_even = {};
+
+  lut_pt_vs_dphi_gemcsc = {
```

Comment:
please add a bit of documentation for these magic numbers (too bad they can't be simply derived from the geometry..)